numpy/_core/src/umath/highway_masked_func.h

```diff
@@ -0,0 +1,46 @@
+#ifndef _NPY_CORE_SRC_UMATH_HIGHWAY_MASKED_FUNC_H_
+#define _NPY_CORE_SRC_UMATH_HIGHWAY_MASKED_FUNC_H_
+
+#include "simd/simd.h"
+#include <hwy/highway.h>
+#include <numpy/npy_common.h>
+
+#if NPY_SIMD
+HWY_BEFORE_NAMESPACE();
+namespace hwy {
+namespace HWY_NAMESPACE {
+
+// Sets `out[idx]` to `Func(d, orig_x[idx])` if `mask[idx]` is true else `transformed_x[idx]`
+// Useful for special casing math routines
+template <class D, class Func, typename T = TFromD<D>>
+HWY_ATTR Vec<D>
+MaskedCallFunc1(D d, Vec<D> orig_x, Vec<D> transformed_x, Mask<D> mask, const Func &fallback)
```

Comment:
The function name doesn't really describe what it actually does. Can we rename to this to something meaningful?

Reply:
Do you have suggestions on naming? I named it this because it is a masked call to a function with one parameter.

Reply:
`MaskedScalarFallback`?